### PR TITLE
switching to use own namespace and make namespace oco

### PR DESCRIPTION
**Pull request by acornett21**
This changes makes the operator be installed in its own namespace via the embedded OperatorHub. This gives the operator a more cleaner instillation documentation.

Signed-off-by: Adam D. Cornett <adc@redhat.com>

### CODE REVIEW

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -215,13 +232,13 @@ spec:
         serviceAccountName: certification-operator-controller-manager
     strategy: deployment
   installModes:
-  - supported: false
+  - supported: true
```
Comment (dcurran90) [inline bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:235]: is this the right move?
from what i understand OLM is moving towards cluster singletons to avoid issues with conflicting installModes
https://github.com/operator-framework/operator-lifecycle-manager/issues/1790
https://groups.google.com/g/operator-framework/c/0KxBa-caG1U

**Comment by itroyano** [comment]
Any changes needed in `main.go`?
`WATCH_NAMESPACE` ?

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (acornett21) [reply bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:235]: I feel like for this operator, since the resources we look for have to be in a specific namespace, and the ones we create are in that same namespace, this is the right approach for now. Maybe we change this once the install docs go away, and we enhance the operator further, but for now with how the operator is coded, and cluster admins have to apply a CatalogSource manually to get it in cluster, I think this is the right direction to be able to get a usable alpha release out.

**Comment by acornett21** [comment]
> Any changes needed in `main.go`? `WATCH_NAMESPACE` ?

In my testing, no nothing in the code needed to be change. All of my resources exist in the `oco` namespace before installing.

File: bundle.Dockerfile [hunk 2]
```diff
@@ -26,7 +26,7 @@ LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=operator-certification-operator
 LABEL operators.operatorframework.io.bundle.channels.v1=alpha
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.13.0
```
Comment (bcrochet) [inline bundle.Dockerfile:29]: This change belongs in its own PR.

File: bundle/manifests/certification.redhat.com_operatorpipelines.yaml [hunk 3]
```diff
@@ -34,6 +34,14 @@ spec:
           spec:
             description: OperatorPipelineSpec defines the desired state of OperatorPipeline
             properties:
+              gitHubSecretName:
```
Comment (bcrochet) [inline bundle/manifests/certification.redhat.com_operatorpipelines.yaml:37]: These changes also seem to be unrelated to changing the namespace settings.

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -107,6 +111,19 @@ spec:
           - patch
           - update
           - watch
+        - apiGroups:
```
Comment (bcrochet) [inline bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:114]: This _seems_ related, but I'm unsure. Can you verify?

File: bundle/manifests/operator-certification-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -233,6 +250,7 @@ spec:
   - email: jomckenz@redhat.com
     name: jomkz
   maturity: alpha
+  minKubeVersion: 1.21.0
```
Comment (bcrochet) [inline bundle/manifests/operator-certification-operator.clusterserviceversion.yaml:253]: Unrelated change.

File: bundle/metadata/annotations.yaml [hunk 6]
```diff
@@ -5,7 +5,7 @@ annotations:
   operators.operatorframework.io.bundle.metadata.v1: metadata/
   operators.operatorframework.io.bundle.package.v1: operator-certification-operator
   operators.operatorframework.io.bundle.channels.v1: alpha
-  operators.operatorframework.io.metrics.builder: operator-sdk-v1.13.0
+  operators.operatorframework.io.metrics.builder: operator-sdk-v1.14.0
```
Comment (bcrochet) [inline bundle/metadata/annotations.yaml:8]: Unrelated change.

File: config/manager/manager.yaml [hunk 7]
```diff
@@ -30,6 +30,7 @@ spec:
         args:
         - --leader-elect
         image: controller:latest
+        imagePullPolicy: Always
```
Comment (bcrochet) [inline config/manager/manager.yaml:33]: Unrelated change.

**Review by bcrochet — CHANGES_REQUESTED** [review]
Lots of unrelated changes.

**Comment by acornett21** [comment]
@bcrochet These changes are because main has changes where `make bundle` was never ran. I'll covert this PR to three PR's. Below are my thoughts.

- Change for `OwnNS`
- Change to run `make bundle` (this PR will have alot of unrelated changes that are here)
- Change for `imagePullPolicy`

**Review by bcrochet — APPROVED** [review]
/lgtm

**Review by jomkz — APPROVED** [review]
